### PR TITLE
Destructo autopromote lmt patch

### PR DESCRIPTION
**Pull request by CaptainDestructo**
This should satisfy the LMTs for the time being until they decide what they want.

### CODE REVIEW

**Comment by hrolfurgylfa** [comment]
Why do the LMT Trainers need to be able to add LMTs with a command? Does the normal process through # request-rank not work for them like it is doing for both regular trainers as well as SLRT?

**Comment by CaptainDestructo** [comment]
They specifically requested to be removed from the feature, as they have a review board that takes place after the #request-rank messages are approved. This command eliminates the need for a Staff member to add the roles, which will make it easier for the LMT trainers to conduct 1-on-1 trainings, or trainings without Foxy present.

**Comment by hrolfurgylfa** [comment]
After discussing it with Foxy, it should be fine to keep them in the feature, they just need some different requirements.

In particular the reviews after trainings only last on average 10-15 minutes, so by having the automatic bot deny time at 1 hour will be fine. Maybe two hours for some extra headroom just in case. This should be fine to be adjusted for everyone, I at least don't see an issue with that being a change for all teams.

After discussing that with her, it seems that it may also be good to add a detector that sees if an X emote is added and remove the request if it detects an X, the X wouldn't necessarily be on the message from the beginning, but that depends on how often a request has to be removed, and we need to talk to more trainers from different teams to get some information on that.

I'm thinking the best way forward is to revert the # request-rank change for now or disable it, and continue looking into it on a separate branch, and then not merge it into main until we show it to leaders of all training teams (LMT, SLRT, ...) and also probably a few extra staff members, just to make completely sure this change is fine, since it changes the # request-rank channel in a fairly big way.

**Comment by CaptainDestructo** [comment]
I’ll look into it further this week

**Comment by CaptainDestructo** [comment]
@hrolfurgylfa Here's the update:

- Readded LMTs to requestables
- Increased timeout to 2 hours default
- Added ❌ reaction hook to delete the message IF the user is a trainer, and the message is in #request-rank

Reopening this pull request, since this branch hasn't been merged yet.

File: Classes/extra_functions.py
Comment (hrolfurgylfa) [inline Classes/extra_functions.py:312]: This will fail if objects[0] is "" and then object[1] actually has something you want to print

File: Classes/extra_functions.py
Comment (hrolfurgylfa) [inline Classes/extra_functions.py:307]: Instead of having lots of these like sep, end, file... we can use **kwargs to collect all the keyword arguments into a dictionary and then convert them back into keyword arguments with **kwargs when you call print. This will also add support for any other keyword arguments print may have.

**Review by hrolfurgylfa — COMMENTED** [review]
The Added timestamps commit should really be a seperate pull request so that we can discuss it separately. I'm thinking that if we're going this far we should probably switch to the [Python logging system](https://docs.python.org/3.8/library/logging.html). Threw some comments on it though in case we want to go with this solution.

**Comment by CaptainDestructo** [comment]
> I'm thinking that if we're going this far we should probably switch to the Python logging system

Utilizing the full logging system is a bit overkill to get just a time stamp, however there are other portions of the code that could use it to display messages.

File: Classes/extra_functions.py
Comment (CaptainDestructo) [reply Classes/extra_functions.py:307]: I try to avoid kwargs when doing functions like this, since it breaks tooltips in most editors.

File: Classes/extra_functions.py
Comment (CaptainDestructo) [reply Classes/extra_functions.py]: Changed to do the “” check only if length is 1. This is to accommodate “” for newline

**Comment by hrolfurgylfa** [comment]
Yeah, I can see that, but this is a pretty big unrelated change though, so go ahead and throw it in a seperate pull request.
-------- Upprunaleg skilaboð --------
Kveikt 5. jún. 2021, 20:59, Daniel Smith skrifaði:

**Comment by CaptainDestructo** [comment]
Will do

**Comment by CaptainDestructo** [comment]
I have moved the commits regarding timestamps into another branch/PR

**Comment by hrolfurgylfa** [comment]
I'll most likely push reviewing this one until tomorrow morning/noon.

**Review by hrolfurgylfa — APPROVED** [review]
Looks good, reviewed in VC

**Pull request merged**
